website/payments/admin_views.py

```diff
@@ -42,3 +50,102 @@ def post(self, request, *args, **kwargs):
             return redirect(request.POST["next"])
 
         return redirect("admin:payments_payment_change", kwargs["pk"])
+
+
+@method_decorator(staff_member_required, name="dispatch")
+@method_decorator(
+    permission_required("payments.process_batches"), name="dispatch",
+)
+class BatchProcessAdminView(View):
+    """
+    View that processes a batch
+    """
+
+    def post(self, request, *args, **kwargs):
+        batch = Batch.objects.get(pk=kwargs["pk"])
+
+        if batch.processed:
+            messages.error(
+                request, _("{} already processed.").format(model_ngettext(batch, 1))
+            )
+        else:
+            batch.processed = True
+            payments = batch.payments_set.select_related("paid_by")
+            for payment in payments:
+                ba = payment.paid_by.bank_accounts.last()
+                ba.last_used = timezone.now()
+                ba.save()
+
+            batch.save()
+            messages.success(
+                request,
+                _("Successfully processed {}.").format(model_ngettext(batch, 1)),
+            )
+
+        if "next" in request.POST:
+            return redirect(request.POST["next"])
+
+        return redirect("admin:payments_batch_change", kwargs["pk"])
+
+
+@method_decorator(staff_member_required, name="dispatch")
+@method_decorator(
+    permission_required("payments.process_batches"), name="dispatch",
```

Comment:
Maybe a bit nitpicky, but since we're exporting IBAN's maybe we should require additional permissions as well

Reply:
There will only be one person doing this: the treasurer. So adding another permission for this would be an immense overkill.